torch/_dynamo/current_scope_id.py

```diff
@@ -0,0 +1,17 @@
+import threading
+
+# Global variable to identify which SubgraphTracer we are in.
+# It is sometimes difficult to find an InstructionTranslator to use.
+_current_scope_id = threading.local()
+
+
+def current_scope_id():
+    global _current_scope_id
+    if not hasattr(_current_scope_id, "value"):
+        _current_scope_id.value = 1
+    return _current_scope_id.value
+
+
+def modify_current_scope_id(delta):
+    global _current_scope_id
+    _current_scope_id.value = current_scope_id() + delta
```

Comment:
No strong opinion, but if we could do this as a context manager, you wouldn't need to invoke `modify_current_scope_id(1)` `modify_current_scope_id(-1)` you could just have this in enter/exit and it would just work.

It would also avoid someone forgetting to call this, or misaligning the delta with a != 1 or -1

Reply:
Hmm, this should be 1:1 with OutputGraph.new_subgraph_tracer, so there should be no need to actually call this outside of that function. That being said, I'll make this a context manager, it will just cause the callsite to look a bit interesting